openshift/e2e-tests-openshift.sh

```diff
@@ -53,33 +54,93 @@ function install_strimzi(){
   sleep 5; while echo && kubectl get pods -n kafka | grep -v -E "(Running|Completed|STATUS)"; do sleep 5; done
 }
 
+function install_servicemesh(){
+  header "Installing ServiceMesh"
+
+  # Install the ServiceMesh Operator
+  oc apply -f https://raw.githubusercontent.com/openshift/knative-serving/release-v0.8.1/openshift/servicemesh/operator-install.yaml
+
+  # Wait for the istio-operator pod to appear
+  timeout_non_zero 900 '[[ $(oc get pods -n openshift-operators | grep -c istio-operator) -eq 0 ]]' || return 1
+
+  # Wait until the Operator pod is up and running
+  wait_until_pods_running openshift-operators || return 1
+
+  # Deploy ServiceMesh
+  oc new-project $SERVICEMESH_NAMESPACE
+  oc apply -n $SERVICEMESH_NAMESPACE -f https://raw.githubusercontent.com/openshift/knative-serving/release-v0.8.1/openshift/servicemesh/controlplane-install.yaml
+  cat <<EOF >> ServiceMeshMemberRoll.yaml
+apiVersion: maistra.io/v1
+kind: ServiceMeshMemberRoll
+metadata:
+  name: default
+  namespace: ${SERVICEMESH_NAMESPACE}
+spec:
+  members:
+  - ${SERVING_NAMESPACE}
+  - ${EVENTING_NAMESPACE}
+EOF
+
+  # process array to create the NS and give SCC
+  testNamesArray=($(cat TEST_NAMES |tr "\n" " "))
+  for i in "${testNamesArray[@]}"
+  do
+  # append test NS members
+ cat <<-EOF >> ServiceMeshMemberRoll.yaml
+- $i
+EOF
+  done
+
+  oc apply -f ServiceMeshMemberRoll.yaml
+
+  # Wait for the ingressgateway pod to appear.
+  timeout_non_zero 900 '[[ $(oc get pods -n $SERVICEMESH_NAMESPACE | grep -c istio-ingressgateway) -eq 0 ]]' || return 1
+
+  wait_until_service_has_external_ip $SERVICEMESH_NAMESPACE istio-ingressgateway || fail_test "Ingress has no external IP"
+  wait_until_hostname_resolves "$(kubectl get svc -n $SERVICEMESH_NAMESPACE istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')"
+
+  wait_until_pods_running $SERVICEMESH_NAMESPACE
+
+  rm ServiceMeshMemberRoll.yaml
```

Comment:
It might be an even better idea to replace ServiceMeshMemberRoll.yaml with a heredoc